### PR TITLE
Fix push with no configured module names

### PR DESCRIPTION
**Pull request by emcfarlane**
This PR allows for better push behaviour when unnamed modules are present. The action will now read the `buf.yaml` config file to check for missing names before running the push or archive steps. With the latest release of `buf`@v.1.33.0 we can now also use the flag `--exclude-unnamed` to avoid failing push on unnamed modules when multiple modules are present. 

As a test (and example) see the [`test-push-unnamed`](https://github.com/bufbuild/buf-action/actions/runs/9503406199/job/26193768827?pr=31) step in CI:
```
/opt/hostedtoolcache/buf/1.33.0/x64/buf push --error-format github-actions --exclude-unnamed --create --git-metadata
WARN	Excluding unnamed module	{"module": "bar"}
buf.build/***/foo:2969b75dd86b4895a5e590dc1bb89
```

In addition is the introduction of outputs in the `action.yml` by setting the resolved version of `buf`.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -249,12 +249,47 @@ jobs:
           push: true
           archive: false
           comment: false
+  push-no-modules:
```
Comment (nicksnyder) [inline .github/workflows/ci.yaml:252]: I think we need to either rename this (and all the other jobs) to make it clear they are e2e tests, or move these tests to their own workflow file. Probably choosing names like `test-push-no-modules` would be fine.

But also, I think the name should be something more like `test-push-no-module-name`

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -313,3 +349,40 @@ jobs:
             label_does_not_exist
             label_does_not_exist_either
           comment: false
+  archive-no-modules:
```
Comment (nicksnyder) [inline .github/workflows/ci.yaml:352]:
```suggestion
  test-archive-no-module-name:
```

File: src/main.ts [hunk 3]
```diff
@@ -108,6 +109,13 @@ async function runWorkflow(bufPath: string, inputs: Inputs): Promise<Steps> {
   if (checks.some((result) => result.status == Status.Failed)) {
     return steps;
   }
+  const modules = parseModules(inputs.input);
+  if (modules.length == 0) {
+    core.info("No modules found, skipping push and archive");
```
Comment (nicksnyder) [inline src/main.ts:114]: there might be modules, they just don't have names to push to the BSR

**Review by nicksnyder — COMMENTED** [review]
Can you add a test that has two modules in the workspace, one with a name and one without?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (emcfarlane) [reply .github/workflows/ci.yaml:252]: All of the CI jobs in the workflow are e2e tests.

Comment (emcfarlane) [reply .github/workflows/ci.yaml]: Moved to `test-` prefix.

File: src/main.ts (hunk 3 above)
Comment (emcfarlane) [reply src/main.ts]: Fixed to `module names`

**Comment by emcfarlane** [comment]
Need to fix multi-module pushes.

File: src/main.ts [hunk 4]
```diff
@@ -252,12 +261,25 @@ async function breaking(bufPath: string, inputs: Inputs): Promise<Result> {
...
     core.info("Skipping push");
     return skip();
   }
+  if (moduleNames.length == 0) {
+    core.info("Skipping push, no named modules detected");
+    return skip();
+  }
   const args = ["push", "--error-format", "github-actions"];
+  // Add --exclude-unnamed on buf versions that support the flag.
+  if (semver.satisfies(bufVersion, ">=1.33.0")) {
```
Comment (nicksnyder) [inline src/main.ts:280]: when we 1.0 this action, I would be in favor of just bumping the required minimum version of the whole action to 1.33.0 so we can eliminate this check

**Review by nicksnyder — APPROVED** [review]

File: src/main.ts (hunk 4 above)
Comment (emcfarlane) [reply src/main.ts:280]: Yep, any version bumps we can increase the min version on. Left here to allow for a patch on 0.1.

**Review by doriable — APPROVED** [review]

**Pull request merged**